File: examples/cpp/Tutorial33.cpp
/* ---------------------------------------------------------------------------
 * Tutorial 33
 * 
 * This tutorial shows how to set document properties for Excel file in C++,
 * like 'Subject' property for summary information, 'Manager' property for
 * document summary information and a custom property.
 * ------------------------------------------------------------------------ */


#include "EasyXLS.h"
#include <conio.h>


int main()
{
	printf("Tutorial 33\n----------\n");

	HRESULT hr;

	// Initialize COM
	hr = CoInitialize(0);

	// Use the SUCCEEDED macro and get a pointer to the interface
	if(SUCCEEDED(hr))
	{
		// Create a pointer to the interface that exports Excel files
		EasyXLS::IExcelDocumentPtr workbook;
		hr = CoCreateInstance(__uuidof(EasyXLS::ExcelDocument),
		NULL,
		CLSCTX_ALL,
		__uuidof(EasyXLS::IExcelDocument),
		(void**) &workbook) ;

		if(SUCCEEDED(hr)){

			// Create a worksheet
			workbook->easy_addWorksheet_2("Sheet1");
			
			// Set the 'Subject' document property
			workbook->getSummaryInformation()->setSubject("This is the subject");

			// Set the 'Manager' document property
			workbook->getDocumentSummaryInformation()->setManager("This is the manager");

			// Set a custom document property
			workbook->getDocumentSummaryInformation()->setCustomProperty("PropertyName", FILEPROPERTY_VT_NUMBER, "4");
			
			// Export Excel file
			printf("Writing file C:\\Samples\\Tutorial33 - Excel file properties.xlsx.");
			workbook->easy_WriteXLSXFile("C:\\Samples\\Tutorial33 - Excel file properties.xlsx");
			
			// Confirm export of Excel file
			_bstr_t sError = workbook->easy_getError();
			if (strcmp(sError, "") == 0){
				printf("\nFile successfully created. Press Enter to Exit...");
			}
			else{
				printf("\nError encountered: %s", (LPCSTR)sError); 
			}
			
			// Dispose memory
			workbook->Dispose();
		}
		else{
			printf("Object is not available!");
		}
	}
	else{
		printf("COM can't be initialized!");
	}

	// Uninitialize COM
	CoUninitialize();

	_getch();
	return 0;
}

File: examples/php-java/Tutorial33.php
<?php require_once("http://localhost:8080/JavaBridge/java/Java.inc");

	/*===========================================================================

	 | Tutorial 33
	 |
	 | This tutorial shows how to set document properties for Excel file in PHP, 
	 | like 'Subject' property for summary information, 'Manager' property for
	 | document summary information and a custom property.
===============================================================================*/

	include("FileProperty.inc");

	header("Content-Type: text/html");

	echo "Tutorial 33<br>";
	echo "----------<br>";

	// Create an instance of the class that exports Excel files
	$workbook = new java("EasyXLS.ExcelDocument");

	// Create a worksheet
	$workbook->easy_addWorksheet("Sheet1");

	// Set the 'Subject' document property
	$workbook->getSummaryInformation()->setSubject("This is the subject");

	// Set the 'Manager' document property
	$workbook->getDocumentSummaryInformation()->setManager("This is the manager");

	// Set a custom document property
	$workbook->getDocumentSummaryInformation()->setCustomProperty("PropertyName", $VT_NUMBER, "4");

	// Export Excel file
	echo "Writing file: C:\Samples\Tutorial33 - Excel file properties.xlsx<br>";
	$workbook->easy_WriteXLSXFile("C:\Samples\Tutorial33 - Excel file properties.xlsx");

	// Confirm export of Excel file
	if ($workbook->easy_getError() == "")
			echo "File successfully created.";
	else
			echo "Error encountered: " .$workbook->easy_getError();

	// Dispose memory
	$workbook->Dispose();	
?>
